tests/test_echo.py

```diff
@@ -145,11 +145,15 @@ async def gen():
             yield echo_pb2.EchoRequest(data=data)
 
     assert (await stub.Echo(echo_pb2.EchoRequest(data=data))).data == data
-    assert [response.data for response in await async_iterable_to_list(
-            stub.EchoTwoTimes(echo_pb2.EchoRequest(data=data)))] == [data] * 2
+    async with stub.EchoTwoTimes(echo_pb2.EchoRequest(data=data)) as aiter:
+        assert [response.data for response in await async_iterable_to_list(aiter)] == [
+            data
+        ] * 2
```

Comment:
Do I understand correctly that it's an API change, even for the case when a generator isn't passed to the call?

I'm not sure about making an API change for aspirational reasons.  We're not using the context, so it doesn't seem worth breaking existing users.

> By requiring the client to enter a context manager before it is able to receive messages from a server stream, it is possible to add functionality in the future to let the server know that the client will never read messages from the stream again whenever it exits the context manager.

Reply:
Yes it is a breaking change. However, I don't consider it to be that problematic, since the breakage is essentially the same as for the case of a stream-stream call. To me it improves the consistency of the API moving forward, even if it is not necessary right now.

Reply:
I think the difference is that any app passing a generator was likely to encounter #36, so likely no one was using the stream-stream feature.  So changing the API is easy to consider.

For this other case, apps would not encounter an error, so it's more likely that someone is using this part of the API, and we're breaking their app.

Reply:
Hmm, that's a fair point. I can remove the change from this PR then. Should the need arise in the future, it can always be changed again.

Reply:
I think limiting the PR to fix the broken case is what I'm comfortable with for now-- thank you.